wolfcrypt/src/port/atmel/atmel.c

```diff
@@ -931,72 +942,131 @@ int atcatls_verify_signature_cb(WOLFSSL* ssl, const byte* sig, unsigned int sigS
+        #define ATCATLS_PUBKEY_BUFF_MAX_SIZE 65
+    #endif
+
     int ret = 0;
     ATCA_STATUS status;
-    size_t signerCertSize=0;
-    size_t deviceCertSize=0;
-    uint8_t *certBuffer;
+    size_t signerCertSize = ATCATLS_SIGNER_CERT_MAX_SIZE;
+    size_t deviceCertSize = ATCATLS_DEVICE_CERT_MAX_SIZE;
+    uint8_t certBuffer[ATCATLS_CERT_BUFF_MAX_SIZE];
+    uint8_t signerBuffer[ATCATLS_SIGNER_CERT_MAX_SIZE];
+#ifdef WOLFSSL_ATECC_TFLXTLS
+    uint8_t signerPubKeyBuffer[ATCATLS_PUBKEY_BUFF_MAX_SIZE];
+#endif
 
-    /* fetch signer cert size */
-    status=tng_atcacert_read_signer_cert(NULL, &signerCertSize);
-    if (ATCA_SUCCESS != status) {
-        #ifdef WOLFSSL_ATECC_DEBUG
-        printf("Failed reading Signer cert size(0x%x)\r\n", status);
-        #endif
-        return WOLFSSL_FAILURE;
+#ifdef WOLFSSL_ATECC_TNGTLS	
+    ret = tng_atcacert_max_signer_cert_size(&signerCertSize);
+    if (ret != ATCACERT_E_SUCCESS) {
+    #ifdef WOLFSSL_ATECC_DEBUG
+       printf("Failed to get max signer cert size\r\n");
+    #endif
+       return ret;
     }
-
-    /* fetch device cert size */
-    status=tng_atcacert_read_device_cert(NULL, &deviceCertSize, NULL);
-    if (ATCA_SUCCESS != status) {
-        #ifdef WOLFSSL_ATECC_DEBUG
-        printf("Failed reading device cert size(0x%x)\r\n", status);
-        #endif
-        return WOLFSSL_FAILURE;
-    }    
-    certBuffer=(uint8_t*)XMALLOC(signerCertSize+deviceCertSize, NULL, DYNAMIC_TYPE_TMP_BUFFER);
-    if(NULL == certBuffer){
-        #ifdef WOLFSSL_ATECC_DEBUG
-        printf("Failed allocating space for certBuffer\r\n");
-        #endif
-        return WOLFSSL_FAILURE;
+    else if (signerCertSize > ATCATLS_SIGNER_CERT_MAX_SIZE) {
+    #ifdef WOLFSSL_ATECC_DEBUG
+        printf("Signer CA cert buffer too small, need to increase at least"
+               " to %d\r\n", signerCertSize);
+    #endif
+       return -1;
     }
-    
-    /* Read signer cert */
-    status = tng_atcacert_read_signer_cert(&certBuffer[deviceCertSize],
-                                           &signerCertSize);
+
+    /* Read TNGTLS signer cert */
+    status = tng_atcacert_read_signer_cert(signerBuffer, &signerCertSize);
     if (ATCA_SUCCESS != status) {
-        #ifdef WOLFSSL_ATECC_DEBUG
-        printf("Error reading signer cert(0x%x)\r\n", status);
-        #endif
-        XFREE(certBuffer,NULL,DYNAMIC_TYPE_TMP_BUFFER);
         ret = atmel_ecc_translate_err(ret);
         return ret;
     }
 
     /* Read device cert signed by the signer above */
     status = tng_atcacert_read_device_cert(certBuffer, &deviceCertSize,
-     	                                   &certBuffer[deviceCertSize]);
+                                           signerBuffer);
     if (ATCA_SUCCESS != status) {
-        #ifdef WOLFSSL_ATECC_DEBUG
-        printf("Error reading device cert(0x%x)\r\n", status);
-        #endif
-        XFREE(certBuffer,NULL,DYNAMIC_TYPE_TMP_BUFFER);
         ret = atmel_ecc_translate_err(ret);
```

Comment:
It looks like places status is returned we should be doing `ret = atmel_ecc_translate_err(status)` or `return atmel_ecc_translate_err(status)`?